### PR TITLE
[onert-micro] Add Concatenation kernel

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This pr adds first Concatenation kernel with common PALUtils.

from draft https://github.com/Samsung/ONE/pull/12426
for issue: https://github.com/Samsung/ONE/issues/12427

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

File: onert-micro/onert-micro/include/pal/mcu/KernelsToBuild.lst [hunk 1]
```diff
@@ -1 +1,3 @@
 REGISTER_KERNEL(ABS, Abs)
+
```
Comment (BalyshevArtem) [inline onert-micro/onert-micro/include/pal/mcu/KernelsToBuild.lst:2]: Place for Add op from #12651

**Review by Torrero — DISMISSED** [review]
LGTM

**Review by SlavikMIPT — DISMISSED** [review]
LGTM

**Comment by BalyshevArtem** [comment]
rebased
@Torrero @SlavikMIPT PTAL)

**Review by SlavikMIPT — APPROVED** [review]

**Review by Torrero — APPROVED** [review]

**Pull request merged**